exec/java-exec/src/test/java/org/apache/drill/TestJoinNullable.java

```diff
@@ -493,4 +496,81 @@ public void withNullEqualAdditionFilter() throws Exception {
         .go();
   }
 
+  @Test
+  public void withMixedEqualAndIsNotDistinctHashJoin() throws Exception {
```

Comment:
Could you add a test with the predicate of type: 
    t1.key = t2.key AND ((t1.data=t2.data) OR (t1.data IS NULL AND t2.data IS NULL))
where the right side of the AND gets converted to IS NOT DISTINCT FROM.